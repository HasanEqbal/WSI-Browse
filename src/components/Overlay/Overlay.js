import React from 'react';
import AlternativeImages from '../AlternativeImages/AlternativeImages';
import AlertInfo from '../AlertInfo/AlertInfo'

export default function Overlay(props) {

  let mainImage = props.product.hero.href;

  let alternativeImages = props.product.images;

  return (
    <div>
      <div className="reveal overlay-container" id="exampleModal1" data-reveal>
        <div className="row align-center">
          <div className="product-image-gallery">
            <img className="pdp-product-image" id="main-product-image" src={mainImage} alt="" />
            <br></br>
            <ul className="menu product-thumbs align-center">
              {alternativeImages ?
                alternativeImages.map((alternativeImage) => {
                  return <AlternativeImages alternativeimage={alternativeImage.href} />
                }) :
                <AlertInfo />}
            </ul>
          </div>
        </div>
      </div>
    </div>
  )

}

