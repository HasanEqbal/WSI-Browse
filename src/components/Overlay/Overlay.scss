@import '../../../src/scss/base/settings';

.product-image-gallery {
  background-color: #fefefe;
  padding: 1.5rem;
}

.pdp-product-image {
  margin-bottom: 20px;
}

.product-thumbs a {
  margin-left: 8px;
  margin-right: 8px;
  padding: 0 !important;
}

.product-thumbs a img {
  display: inline-block;
  max-width: 100%;
  margin-bottom: 1rem;
  border: solid 4px #fefefe;
  border-radius: 0;
  box-shadow: 0 0 0 1px rgba(10, 10, 10, 0.2);
  line-height: 0;
  border-radius: 3px;
  width: 50px;
  height: 50px;
  border: none;
}
.overlay{
  position: fixed;
  width: 40%;
  height: 40%;
  margin-left: 18%;
  background-color: #f1c40f;
  text-align: center;


  z-index: 10;
  outline: 9999px solid rgba(0,0,0,0.5);

}

.awssld {
  --slider-height-percentage: 90% !important;
  --control-bullet-color: #1468a0 !important;
  --control-bullet-active-color: #1468a0 !important;
}

.close-button{
  display:flex;
  justify-content:flex-end;
  width:100%;
  padding:0;
}

@media only screen and (max-width: 600px) {
  .overlay{
  position: fixed;
  width: 80%;
  height: 80%;
  background-color: #f1c40f;
  text-align: center;
  margin-left: 2%;
  margin-top: 2%;
  }

  .awssld {
    --slider-height-percentage: 90% !important;
    --control-bullet-color: #1468a0 !important;
    --control-bullet-active-color: #1468a0 !important;
  }

}
