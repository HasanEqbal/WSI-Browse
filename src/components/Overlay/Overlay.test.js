import React from 'react';
import { shallow } from 'enzyme';
import Overlay from './Overlay';


describe('Component : Overlay', () => {

  it('expect to render Overlay component', () => {
    expect(shallow(<Overlay />)).toMatchSnapshot();
  })

})
