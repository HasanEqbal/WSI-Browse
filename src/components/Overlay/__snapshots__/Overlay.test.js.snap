// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component : Overlay expect to render Overlay component 1`] = `ShallowWrapper {}`;
